sdk-extensions/aws/src/main/java/io/opentelemetry/sdk/extension/aws/trace/GetSamplingRulesResponse.java

```diff
@@ -0,0 +1,173 @@
+
+    abstract GetSamplingRulesResponse build();
+  }
+
+  @AutoValue
+  @JsonDeserialize(builder = SamplingRuleRecord.Builder.class)
+  abstract static class SamplingRuleRecord {
+
+    @JsonProperty("CreatedAt")
+    abstract String getCreatedAt();
+
+    @JsonProperty("ModifiedAt")
+    abstract String getModifiedAt();
+
+    @JsonProperty("SamplingRule")
+    abstract SamplingRule getRule();
+
+    @AutoValue.Builder
+    abstract static class Builder {
+
+      @JsonCreator
+      static Builder builder() {
+        return new AutoValue_GetSamplingRulesResponse_SamplingRuleRecord.Builder();
+      }
+
+      @JsonProperty("CreatedAt")
+      abstract Builder setCreatedAt(String createdAt);
+
+      @JsonProperty("ModifiedAt")
+      abstract Builder setModifiedAt(String modifiedAt);
+
+      @JsonProperty("SamplingRule")
+      abstract Builder setRule(SamplingRule rule);
+
+      abstract SamplingRuleRecord build();
+    }
+  }
+
+  @AutoValue
+  @JsonDeserialize(builder = SamplingRule.Builder.class)
+  abstract static class SamplingRule {
+    @JsonProperty("Attributes")
+    abstract Map<String, String> getAttributes();
+
+    @JsonProperty("FixedRate")
+    abstract double getFixedRate();
+
+    @JsonProperty("Host")
+    abstract String getHost();
+
+    @JsonProperty("HTTPMethod")
+    abstract String getHttpMethod();
+
+    @JsonProperty("Priority")
+    abstract int getPriority();
+
+    @JsonProperty("ReservoirSize")
+    abstract int getReservoirSize();
+
+    @JsonProperty("ResourceARN")
+    abstract String getResourceArn();
+
+    @JsonProperty("RuleARN")
+    @Nullable
+    abstract String getRuleArn();
+
+    @JsonProperty("RuleName")
+    @Nullable
+    abstract String getRuleName();
+
+    @JsonProperty("ServiceName")
+    abstract String getServiceName();
+
+    @JsonProperty("ServiceType")
+    abstract String getServiceType();
+
+    @JsonProperty("URLPath")
+    abstract String getUrlPath();
+
+    @JsonProperty("Version")
+    abstract int getVersion();
+
+    @AutoValue.Builder
+    abstract static class Builder {
+
+      @JsonCreator
+      static Builder builder() {
+        return new AutoValue_GetSamplingRulesResponse_SamplingRule.Builder();
+      }
+
+      @JsonProperty("Attributes")
+      abstract Builder setAttributes(Map<String, String> attributes);
+
+      @JsonProperty("FixedRate")
+      abstract Builder setFixedRate(double fixedRate);
+
+      @JsonProperty("Host")
+      abstract Builder setHost(String host);
+
+      @JsonProperty("HTTPMethod")
+      abstract Builder setHttpMethod(String httpMethod);
+
+      @JsonProperty("Priority")
+      abstract Builder setPriority(int priority);
+
+      @JsonProperty("ReservoirSize")
+      abstract Builder setReservoirSize(int reservoirSize);
+
+      @JsonProperty("ResourceARN")
+      abstract Builder setResourceArn(String resourceArn);
+
+      @JsonProperty("RuleARN")
+      @Nullable
```

Comment:
You don't need to annotate the method with `@Nullable` since it returns the builder that won't be null

Reply:
Oops thanks!